### PR TITLE
Update to v1.0.0 alpha.4

### LINKED ISSUE #61: Update to alpha4
https://github.com/starkware-libs/cairo/releases/tag/v1.0.0-alpha.4

### PR DESCRIPTION
**Pull request by trangnv**
Close #61 
Time spend: 3 hours

Breaking changes that affect some exercises
- [ ] Felts comparison operators `<, >, <=, >=, !=` are not supported anymore
Exercise affected: `primitive_types2`, `operations1`, `if1`, `quiz1`, `options1`, `structs3`
- [x] Error in `get_gas()` in `options3`, ~~haven't checked what is it exactly~~. Fixed

The solution for now is to comment those exercises out and open a separated issue to fix them later, 1 possibility is to replace felts by integers

### CODE REVIEW

**Comment by msaug** [comment]
> Felts comparison operators <, >, <=, >=, != are not supported anymore
Exercise affected: primitive_types2, operations1, if1, quiz1, options1, structs3


If this is the direction the language is taking (removing native felt operations so that developers don't use felts anymore), then let's adapt starklings in this direction as well 😄

**Comment by trangnv** [comment]
That makes a lot of sense imo.
But what is the solution for string? Here we have exercise relies on felt-string comparisons.

**Comment by trangnv** [comment]
`primitive_types2` and `structs3`

**Comment by trangnv** [comment]
Wait, only `primitive_types2`.

But checking back `structs3`, should we have `#[should_panic]` for  the test `fn fail_creating_weightless_package()`?

**Comment by trangnv** [comment]
Now the only one that is unsolved is `primitive_types2` with strings comparisons, others are good with alpha.4

File: corelib/Scarb.toml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[package]
+name = "core"
+version = "1.0.0-alpha.3"
```
Comment (msaug) [inline corelib/Scarb.toml:3]: alpha 4

File: exercises/operations/operations1.cairo [hunk 2]
```diff
@@ -1,31 +1,32 @@
-// Felts accepted +, -, * but not /
-// Beside that, felts also support comparison operators
-// <, >, <=, >=, ==, !=
-// Fill in the rest of the line that has code missing!
+// Compare to Cairo0, Cairo1 provide integer types and its operations including comparisons
+// Felts are now abstracting away
```
Comment (msaug) [inline exercises/operations/operations1.cairo:2]: This could be more precise.

File: exercises/structs/structs3.cairo [hunk 3]
```diff
@@ -11,17 +11,17 @@ use array::ArrayTrait;
...
-    weight_in_grams: felt,
+    weight_in_grams: usize,
 }
 
 trait PackageTrait {
-    fn new(sender_country: felt, recipient_country: felt, weight_in_grams: felt) -> Package;
-    fn is_international(ref self: Package) -> //???;
-    fn get_fees(ref self: Package, cents_per_gram: felt) -> //???;
+    fn new(sender_country: felt, recipient_country: felt, weight_in_grams: usize) -> Package;
+    fn is_international(ref self: Package) -> bool; //???;
```
Comment (msaug) [inline exercises/structs/structs3.cairo:19]: remove types here

File: exercises/options/options1.cairo [hunk 4]
```diff
@@ -17,18 +17,18 @@ fn maybe_icecream(time_of_day: felt) -> Option::<felt> {
...
+    assert(maybe_icecream(22_usize).unwrap() == 0_usize, 'err_4');
+    assert(maybe_icecream(25_usize).is_none(), 'err_5');
 }
 
 #[test]
 fn raw_value() {
     // TODO: Fix this test. How do you get at the value contained in the Option?
-    let icecreams = maybe_icecream(12);
-    assert(icecreams == 5, 'err_6');
+    let icecreams = maybe_icecream(12_usize).unwrap();
```
Comment (msaug) [inline exercises/options/options1.cairo:30]: remove unwrap

**Comment by msaug** [comment]
Felt comparison was actually removed in a commit after alpha4. Let's keep it for now, hopefully alpha5 will come with a type for felt-string comparison

**Review by msaug — APPROVED** [review]
I applied the last fixes. Looks good now, thanks Trang :)

**Pull request merged**